### ISSUE TITLE
Missing single quote character in date encoding 

### ISSUE THREAD

**Issue by prashantbarca** [michaelrsweet/htmldoc#455]
https://github.com/michaelrsweet/htmldoc/blob/master/htmldoc/ps-pdf.cxx#L11539

```
fprintf(out, "%%%%CreationDate: D:%04d%02d%02d%02d%02d%02d+0000\n",
```

Both PDF 1.7 and 2.0 specifications require the date to take the following form `(D:YYYYMMDDHHmmSSOHH'mm)`, with an additional single quote in the end optional.

**Comment by michaelrsweet** [comment]
@prashantbarca That is for the PostScript output, which doesn't use the PDF date/time format.

The code later in the same function is responsible for the PDF CreationDate value, but had the same issue.

(FWIW, older versions of the PDF specification used a slightly different format for date/time values, yay...)

**Comment by michaelrsweet** [comment]
[master 7c93d8a] Fix the PDF CreationDate format (Issue #455)

**Comment by petervwyatt** [comment]
Let's clarify: only the original Adobe PDF 1.0 spec left the date format completely unspecified. 
In Adobe PDF 1.1, they introduced the "(D:YYYYMMDDHHmmSSOHH'mm')" form with the trailing quote. 
And the only thing that changed between 1996 and now is that the trailing quote is now recognized as optional due to previous spec typos. The latest ISO 32000-2:2020 standard also clarified and refined the understanding of what comprises a valid date string (i.e. they do not have to always be literal strings): "A date string can be any valid PDF string object as described in 7.3.4, "String objects". The description above relates to the text string value after appropriate processing." - so dates can be literal strings with octal escapes, hex strings, etc. 
But the fix makes things good and explicit (if everything is really UT).
